app/crud.py

```diff
@@ -0,0 +1,35 @@
+from sqlalchemy.orm import Session
+from . import models, schemas
+
+def get_city(db: Session, city_id: int):
+    return db.query(models.City).filter(models.City.id == city_id).first()
+
+def get_cities(db: Session):
+    return db.query(models.City).all()
+
+def create_city(db: Session, city: schemas.CityCreate):
+    db_city = models.City(name=city.name, additional_info=city.additional_info)
+    db.add(db_city)
+    db.commit()
+    db.refresh(db_city)
+    return db_city
+
+def delete_city(db: Session, city_id: int):
+    db_city = db.query(models.City).filter(models.City.id == city_id).first()
+    if db_city:
+        db.delete(db_city)
+        db.commit()
+    return db_city
+
+def get_temperatures(db: Session, city_id: int = None):
+    query = db.query(models.Temperature)
+    if city_id:
+        query = query.filter(models.Temperature.city_id == city_id)
+    return query.all()

```

Comment:
The `get_temperatures` function could benefit from handling cases where no temperatures are found for a given `city_id`. Although returning an empty list is acceptable, logging or notifying the user might be helpful in some contexts.